### ISSUE TITLE
19-Diffusion Self-Guidance for Controllable Image Generation

### ISSUE THREAD

**Issue by mjmaher987** [sharif-ml-lab/visual_compositional_generation_survey#20]
## Task: Read the Paper and Do the following 4 steps
1. Highlight the paper systematically (**green** for important notes - **blue** for very important notes that worth mentioning in the final survey - **red** for the sentences that we don’t understand or need further attention in the future) [In Zotero].
2. Write the Wikipedia summary of the paper in the GitHub issue (_comment in this issue_).
3. Extract the information in the [excel file](https://docs.google.com/spreadsheets/d/15_sh6TqtUvZIJo1-RZxQtsG0Nvg0vPG_IFKCWiGMjdY/edit?usp=sharing).
4. Add the related sentences to the [overleaf file](https://www.overleaf.com/8167246262ccxtrzztjvpz). Note that when you add the sentence, don’t forget to write the reference in the format of ”[ #reference number that is mentioned in the Google Doc - Name of the Author (who added this to the overleaf) ]” at the end of the sentence.

**Comment by geekn0rd** [comment]
### **Introduction and Objective:**
Large-scale generative T2I models are capable of producing high-quality images from a detailed text description. However, many properties of an image are difficult to convey through text, in addition there's no option for simple edits. For example, how can we move or resize an object? How can we take the appearance of an object in one image and copy it over to another? etc.
In this paper authors introduce self-guidance, a method that provides greater control over generation by guiding the internal representations of diffusion models. They show that properties such as the shape, location, and appearance of objects can be extracted from these representations.



###  **Methodology:**

The method presented is based on classifier free guidance. It wants to introduce new score functions that can control properties of objects and interactions between them, and then use it to generate the desired image.
**Where can we find signal for controlling diffusion?**
Recent work has shown that the intermediate outputs of the diffusion U-Net encode valuable information about the structure and content of the generated images. In particular, the self and cross-attention maps $`\{A_{i,t} \in R^{H_i×W_i×K} \}`$	 often encode structural information about object position and shape, while the network activations $`\{ \Psi_{i,t} \in R^{H_i×W_i×D_i} \}`$	 allow for maintaining coarse appearance when extracted from appropriate layers.




###  **Contributions:**
- Self-Guidance, which takes advantage of the internal representations of pre-trained T2I diffusion models to provide disentangled, zero-shot control over the generative process without requiring auxiliary models or supervision. 

-  Properties such as the size, location, shape, and appearance of objects can be extracted from these representations and used to meaningfully guide sampling in a zero-shot manner. 
- Small set of properties, when composed, allows for a wide variety of surprisingly complex image manipulations.
-  This method can be extended to editing real images.

###  **Experiments/Results:**
- Relying on cross-attention maps in a model imposes limitations, preventing control over objects not mentioned in the conditioning text prompt and hindering fully disentangled control between interacting objects. The authors propose that selectively applying attention guidance at specific layers or timesteps could enhance disentanglement. Experiments indicate that appearance features often contain unwanted spatial layout information, possibly due to the model's access to positional information. Conversely, guiding the shape of tokens can reveal object appearances, implying that hidden patterns from attention channel interactions may encode appearance. The findings propose the method as a tool to understand diffusion models, offering experimental insights for future research.
### **Comparison:**

###  **Conclusion and Future Work:**
- The paper highlights transformative use-cases of a certain technology but acknowledges the risk of generating harmful content with potential negative societal impacts. Specifically, the ability for self-guidance in the generation process raises concerns about the potential creation of harmful image manipulations, such as incorporating real images' appearance or layout into arbitrary generated content. The example provided involves injecting a public figure into an image depicting illicit activity. To address these risks, the experiments deliberately avoid generating images containing humans. The paper suggests additional safeguards, such as embedded watermarking and automated systems for filtering generated imagery, to mitigate the potential harm.
